Adding Support for @LinkMenu to @TreeGrid

lopezton-anthem opened the pull request:
# Description
<!-- Include a summary of the change and which issue is fixed. Also include relevant motivation and context. List any dependencies that are required for this change. -->

- Added Support for @LinkMenu to @TreeGrid

# Overview of Changes
<!-- Please include a bulleted list of changes here -->

- Added Support for @LinkMenu to @TreeGrid
- Combined similar logic from table, grid, and tree grid into a single base class for using `@LinkMenu` component.

# Type of Change
<!-- Please include the options below which are relevant. -->
<!--
- [ ] Bug fix
- [ ] Refactor/Technical Debt
- [ ] Documentation Update
- [ ] Test case change
- [ ] This change requires a documentation update
- [ ] Other
-->

- [ ] New feature

# Test Details
<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. -->

Functional testing in Petclinic

# Additional Notes
<!-- Please add any additional notes regarding this pull request here. -->

N/A

openanthem-admin commented:
You did it @lopezton-anthem!

Thank you for signing the Anthem Opensource Software CLA.

lopezton-anthem commented:
Two Changes:
 - @sandeepmantha I have added a story in Jira ([NIM-20469](https://jira.anthem.com/browse/NIM-20469))
 - @dinakarmeda I have removed `grid.component.ts` from the source code.
